packages/components/src/Popover/Popover.tsx

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable no-null/no-null */
+import React, { useState } from "react";
+import { usePopper } from "react-popper";
+import classnames from "classnames";
+import classes from "./Popover.css";
+import { Typography } from "../Typography";
+import { ButtonDismiss } from "../ButtonDismiss";
+
+export interface PopoverProps {
+  /**
+   * Element the popover will attach to and point at.
+   */
+  readonly attachTo: Element | React.RefObject<Element | null>;
+
+  /**
+   * Pop-over content.
+   */
+  readonly children: React.ReactNode;
+
+  /**
+   * Whether or not the popover is dismissable via the header close button.
+   */
+  readonly dismissible?: boolean;
+
+  /**
+   * Control popover viability.
+   */
+  readonly open?: boolean;
+
+  /**
+   * Callback executed when the user wants to close/dismiss the popover
+   */
+  readonly onRequestClose?: () => void;
+
+  /**
+   * Describes the preferred placement of the popper.  The first element in the placement array will be taken
+   * as the preferred location with the rest of the array as fallback.
+   */
+  readonly preferredPlacement?: "top" | "bottom" | "left" | "right" | "auto";
```

Comment:
Since `auto` is the permanent fallback, should it still be a prop option?

Reply:
wouldn't typescript complain if we didn't have `auto` as a type when we use it as a fallback?

Reply:
It is nice if the default is also available as an option so I think it's good to include here. 

Would also be nice to tag the default here. Eg.
https://github.com/GetJobber/atlantis/blob/master/packages/components/src/Avatar/Avatar.tsx#L27-L29